test_app/tests/test_fields.py

```diff
@@ -34,4 +43,94 @@ def test_encrypted_field_returns_full_descrypted_value(self):
 
         encrypted_return = EncryptedClassNoDisplayChars.objects.get(id=encrypted_class.id)
 
-        self.assertEqual(value, encrypted_return.encrypted_val)
\ No newline at end of file
+        self.assertEqual(value, encrypted_return.encrypted_val)
+
+    def test_json_text_field_accepts_json_value(self):
+        value = '123456789'
+        encrypted_class = DataGenerator.set_up_encrypted_class(value=value)
+
+        encrypted_return = EncryptedClass.objects.get(id=encrypted_class.id)
+
+        self.assertEqual(value[-4:], encrypted_return.encrypted_val[-4:])
+        self.assertEqual(len(encrypted_return.encrypted_val), 4)
+
+    class JsonTextFieldTests(unittest.TestCase):
+
+        @patch.object(SimplifyJsonTextField, 'to_python')
+        def test_json_text_field_calls_mock_to_python_when_set(self, mock_to_python):
+            # arrange
+            mock_to_python.return_value = []
+            json_text = json.loads('[{ "description": "isn\'t it beautiful outside?" }]')
+
+            # act
+            jt_class = DataGenerator.set_up_json_text_field_class(
+                json_text=json_text)
+
+            # assert
+            self.assertTrue(mock_to_python.called)
+
+        @patch.object(SimplifyJsonTextField, 'from_db_value')
+        def test_json_text_field_calls_mock_from_db_value_when_retrieved_from_db(self, mock_from_db):
+            # arrange
+            mock_from_db.return_value = []
+            json_text = json.loads('[{ "description": "isn\'t it beautiful outside?" }]')
+            jt_class = DataGenerator.set_up_json_text_field_class(
+                json_text=json_text)
+
+            # act
+            jt_return = JsonTextFieldClass.objects.get(id=jt_class.id)
+
+            # assert
+            self.assertTrue(mock_from_db.called)
+
+        def test_json_text_field_returns_json_value_as_list(self):
+            # arrange
+            json_text = json.loads('{ "description": "isn\'t it beautiful outside?" }')
+            jt_class = DataGenerator.set_up_json_text_field_class(
+                json_text=json_text)
+
+            # act
+            jt_return = JsonTextFieldClass.objects.get(id=jt_class.id)
+
+            # assert
+            self.assertIsNotNone(jt_return)
+            self.assertTrue(jt_return.json_text, List)
```

Comment:
Sorry I meant why does this return a list and not a dict, the json seems like a dict not a list.

Reply:
Great catch dude!! I screwed those tests up and they still passed because I failed to add `isinstance()` to them. You were right though. I initially set this up as a list but then in adding more tests it must have gotten changed somehow.

Reply:
I've since fixed them all to assert as expected.